src/main/java/com/google/sps/servlets/TasksServlet.java

```diff
@@ -45,26 +42,25 @@
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // get Google credential object. Ensure it is valid - otherwise return an error to client
-    Credential googleCredential = AuthenticationUtility.getGoogleCredential(request);
-    if (googleCredential == null) {
-      response.sendError(403, AuthenticationUtility.ERROR_403);
-      return;
-    }
+    // Load and verify google credential
+    super.doGet(request, response);
 
-    // Get tasks from Google Tasks
-    Tasks tasksService = TasksUtility.getTasksService(googleCredential);
-    List<TaskList> taskLists = TasksUtility.listTaskLists(tasksService);
-    List<Task> tasks = new ArrayList<>();
-    for (TaskList taskList : taskLists) {
-      tasks.addAll(TasksUtility.listTasks(tasksService, taskList));
-    }
+    // Credential is null if user is not authenticated.
+    if (googleCredential != null) {
+      // Get tasks from Google Tasks
+      Tasks tasksService = TasksUtility.getTasksService(googleCredential);
+      List<TaskList> taskLists = TasksUtility.listTaskLists(tasksService);
+      List<Task> tasks = new ArrayList<>();
+      for (TaskList taskList : taskLists) {
+        tasks.addAll(TasksUtility.listTasks(tasksService, taskList));
+      }
```

Comment:
Would be nice to wrap this in a helper function!

Reply:
Agreed - done!